httpclient/response.go

```diff
@@ -71,3 +83,94 @@ func WithResponseUnmarshal(response any) DoOption {
 		},
 	}
 }
+
+func tryInjectContent(response any, body *common.ResponseWrapper) bool {
+	value := reflect.ValueOf(response)
+	value = reflect.Indirect(value)
+	if value.Kind() != reflect.Struct {
+		return false
+	}
+
+	// Check if there is a "Contents" of typo io.ReadCloser
+	// This is by internal convention with the teams.
+	ioType := reflect.TypeOf((*io.ReadCloser)(nil)).Elem()
+	contentField := value.FieldByName("Contents")
+	if !contentField.IsValid() || !contentField.CanSet() || contentField.Type() != ioType {
+		return false
+	}
+	// If so, set the value
+	contentField.Set(reflect.ValueOf(body.ReadCloser))
+	return true
+}
+
+func injectHeaders(response any, body *common.ResponseWrapper) error {
+	value := reflect.ValueOf(response)
+	value = reflect.Indirect(value)
+	objectType := value.Type()
+
+	if objectType.Kind() != reflect.Struct {
+		return nil
+	}
+	headers := body.Response.Header
+
+	for i := 0; i < objectType.NumField(); i++ {
+		field := objectType.Field(i)
+		headerTag := parseHeaderTag(field)
+		if headerTag.ignore {
+			continue
+		}
+		// We only support a single value for a header, and of types int and string.
+		headerValue := headers.Get(headerTag.name)
+		if headerValue == "" {
+			continue
+		}
+		switch field.Type.Kind() {
+		case reflect.String:
+			value.Field(i).Set(reflect.ValueOf(headerValue))
+		case reflect.Int:
+			intValue, err := strconv.Atoi(headerValue)
+			if err != nil {
+				return fmt.Errorf("failed to parse header %s as int: %w", headerTag.name, err)
+			}
+			value.Field(i).Set(reflect.ValueOf(intValue))
+		default:
+			// Don't fail the request if we can't inject a header for backwards compatibility.
+			logger.Warnf(context.Background(), "unsupported header type %s for field %s", field.Type.Kind(), field.Name)
```

Comment:
Can fail

Reply:
What happens if there is a different type of header (other than int, string)? Should we fail earlier? If we don't fail and headers aren't injected properly (which are further used downstream) then this could lead to incorrect behaviour right?

Reply:
Each option has downsides. Let's have a discussion here.